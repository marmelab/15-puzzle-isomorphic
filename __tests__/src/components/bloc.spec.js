import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Bloc from '../../../src/components/bloc';

describe('Bloc', () => {
    test('should render Bloc without error', () => {
        renderer.create(
            <Bloc>
                <div>Hello</div>
            </Bloc>,
        );
    });

    test('should render the Bloc correctly', () => {
        const component = renderer.create(
            <Bloc>
                <div>Hello</div>
            </Bloc>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should render the title prop if passed', () => {
        const app = shallow(
            <Bloc title="title">
                <div>Hello</div>
            </Bloc>,
        );
        expect(app.find('h5').text()).toEqual('title');
    });

    test('should display a loader', () => {
        const app = shallow(
            <Bloc title="title" isLoading={true}>
                <div>Hello</div>
            </Bloc>,
        );

        expect(app.find('.activity-indicator-wrapper').length).toEqual(1);
    });
});
