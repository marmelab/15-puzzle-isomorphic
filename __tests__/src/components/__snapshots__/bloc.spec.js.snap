// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bloc should render the Bloc correctly 1`] = `
<div
  className="bloc z-depth-3"
>
  
  <div
    className="col s12"
  >
    <div>
      Hello
    </div>
  </div>
</div>
`;
